### PR TITLE
fix: Remove social media

### PR DESCRIPTION
**Pull request by alainajaffer**

### CODE REVIEW

File: site/pages/api/delete-post.api.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { DISRUPTION_DETAIL_PAGE_PATH, REVIEW_DISRUPTION_PAGE_PATH } from "../../constants";
+import { removeSocialMediaPostFromDisruption, upsertSocialMediaPost } from "../../data/dynamo";
+import { SocialMediaPostTransformed as SocialMediaPost } from "../../schemas/social-media.schema";
+import { redirectTo, redirectToError } from "../../utils/apiUtils";
+import { getSession } from "../../utils/apiUtils/auth";
+
+const deleteConsequence = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
```
Comment (alvak1) [inline site/pages/api/delete-post.api.ts:8]: Update the variable name to remove consequence

File: site/data/dynamo.ts [hunk 2]
```diff
@@ -516,6 +562,20 @@ export const removeConsequenceFromDisruption = async (index: number, disruptionI
     );
 };
 
+export const removeSocialMediaPostFromDisruption = async (index: number, disruptionId: string, id: string) => {
+    logger.info(`Updating socialMediaPost ${index} in disruption (${disruptionId}) in DynamoDB table...`);
```
Comment (alvak1) [inline site/data/dynamo.ts:566]: Comment needs to be updated to indicate that the social media post is deleted and not updated

File: site/pages/api/delete-post.api.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            redirectToError(res, message, "api.delete-socialMediaPost", e);
+            return;
+        }
+
+        redirectToError(res);
+        return;
+    }
+};
+
+export default deleteConsequence;
```
Comment (alvak1) [inline site/pages/api/delete-post.api.ts:69]: Update the variable name to remove consequence

File: site/pages/api/delete-post.test.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    afterEach(() => {
+        vi.resetAllMocks();
+    });
+
+    afterAll(() => {
+        MockDate.reset();
+    });
+
+    it("should retrieve valid data from cookies, write to dynamo and redirect", async () => {
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/api/delete-post.test.ts]: done

File: site/pages/api/delete-post.api.ts (hunk 3 above)
Comment (alainajaffer) [reply site/pages/api/delete-post.api.ts]: done changed to post

File: site/data/dynamo.ts (hunk 2 above)
Comment (alainajaffer) [reply site/data/dynamo.ts]: done

File: site/pages/api/delete-post.api.ts (hunk 1 above)
Comment (alainajaffer) [reply site/pages/api/delete-post.api.ts]: done changed to post

File: site/pages/api/delete-post.test.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        upsertSocialMediaPost: vi.fn(),
+    }));
+
+    vi.mock("crypto", () => ({
+        randomUUID: () => "id",
+    }));
+
+    MockDate.set("2023-03-03");
+
+    const deleteConsequenceSpy = vi.spyOn(dynamo, "removeSocialMediaPostFromDisruption");
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/api/delete-post.test.ts]: changed to post

**Review by alvak1 — APPROVED** [review]
Approved

**Pull request merged**